src/java/picard/analysis/artifacts/Transition.java

```diff
@@ -51,4 +46,46 @@ public Transition matchingRef() {
     public String toString() {
         return this.ref + ">" + this.call;
     }
+
+    protected enum Base {
+        A ('A'),
+        C ('C'),
+        G ('G'),
+        T ('T');
+
+        public byte base;
+
+        private Base(final char base) {
+            this.base = (byte)base;
+        }
+    }
+
+    // only 4 of the values will be used but we want to optimize for speed by accessing via the int value of a char
+    static protected final int[] baseIndexMap = new int[256];
+    static {
+        Arrays.fill(baseIndexMap, -1);
+        for (final Base b : Base.values()) {
+            baseIndexMap[b.base] = b.ordinal();
+        }
+    }
+
+    static private final Transition[][] transitionIndexMap = new Transition[Base.values().length][Base.values().length];
+    static {
+        transitionIndexMap[Base.A.ordinal()][Base.A.ordinal()] = AtoA;
```

Comment:
I'll send you over hipchat a file to tie out with. OK?



On Mon, Dec 21, 2015 at 11:19 AM, Eric Banks <notifications@github.com>
wrote:

> In src/java/picard/analysis/artifacts/Transition.java
> <https://github.com/broadinstitute/picard/pull/363#discussion_r48161240>:
>
> > +            this.base = (byte)base;
> > +        }
> > +    }
> > +
> > +    // only 4 of the values will be used but we want to optimize for speed by accessing via the int value of a char
> > +    static protected final int[] baseIndexMap = new int[256];
> > +    static {
> > +        Arrays.fill(baseIndexMap, -1);
> > +        for (final Base b : Base.values()) {
> > +            baseIndexMap[b.base] = b.ordinal();
> > +        }
> > +    }
> > +
> > +    static private final Transition[][] transitionIndexMap = new Transition[Base.values().length][Base.values().length];
> > +    static {
> > +        transitionIndexMap[Base.A.ordinal()][Base.A.ordinal()] = AtoA;
>
> @yfarjoun <https://github.com/yfarjoun> I tried it but you can't use
> valueOf in the static initializer...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/363/files#r48161240>.
>